src/Daemon/Daemon.cpp

```diff
@@ -460,6 +468,64 @@ namespace usbguard
   {
   }
 
+  void Daemon::daemonize(const std::string &pid_file)
+  {
+      pid_t pid = 0;
+      int fd;
+
+      pid = fork();
+      if (pid < 0) {
+          USBGUARD_LOG(Error) << "Fork failed with " << pid << ", exiting.";
+          exit(EXIT_FAILURE);
+      }
+      if (pid > 0) {
+          exit(EXIT_SUCCESS);
+      }
+
+      /* Now we are forked */
+      if (setsid() < 0) {
+          exit(EXIT_FAILURE);
+      }
+      signal(SIGCHLD, SIG_IGN);
+
+
+      pid = fork();
+      if (pid < 0) {
+          USBGUARD_LOG(Error) << "Fork failed with " << pid << ", exiting.";
+          exit(EXIT_FAILURE);
+      }
+      if (pid > 0) {
+          exit(EXIT_SUCCESS);
+      }
+
+      /* Now we are forked 2nd time */
+      umask(0);
+      chdir("/");
+      for (fd = sysconf(_SC_OPEN_MAX); fd > 0; fd--) {
```

Comment:
Are you sure this doesn't also close the file descriptors used by the logger? i.e. the syslog socket, log file fd, ... if so, we need to adapt one thing or the other.